src/Makefile

```diff
@@ -509,6 +535,9 @@ compile_flags.txt:
 
 luacartesi: cartesi.so cartesi/grpc.so cartesi/jsonrpc.so
 
+compute-uarch-pristine-hash: $(COMPUTE_UARCH_PRISTINE_HASH_OBJS)
```

Comment:
Actually, I am not convinced we need this `uarch-pristine-state-hash` extra compile step. I think we could auto generate this hash on the fly the first time it is requested based on the `uarch_pristine_ram`.

Furthermore, having a hard-coded hash is making the `--uarch-ram-image` misbehave I think.

Reply:
The machine-solidity version of the `reset_uarch` code does not have access to an emulator, so it needs to include a hard-coded hash. Since it will auto-generated from the corresponding emulator's cpp code, it makes sense for that code to include the hard-coded value as well. This will evolve a bit once Stephen is working on it.

Reply:
Makes sense, ignore the suggestion then. But should we also remove --uarch-ram option? Since its hash will not match the hard-coded one.

Reply:
Not really, because we use it to test the implementation of the uarch itself.